src/rpc/rpc.cc

```diff
@@ -1143,6 +1143,16 @@ future<> server::connection::send_unknown_verb_reply(std::optional<rpc_clock_typ
       });
   }
 
+  future<> server::connection::abort_all_streams() {
+      return parallel_for_each(_streams | boost::adaptors::map_values, [] (xshard_connection_ptr s) {
+          return smp::submit_to(s->get_owner_shard(), [s] {
+              s->get()->abort();
+          });
+      }).then([this] {
+          _streams.clear();
+      });
```

Comment:
Could there be some strange race here?
1. some operation begins that would eventually add an entry to _streams
2. we stop the server
3. the operation completes just before _streams.clear()

I think it's still better than what we have now, even with the race.